release.sh

```diff
@@ -1,30 +1,30 @@
 #!/bin/sh
 
-read -p "Enter new version number: " version
+version="$1"
+if [ -z "$version" ]; then
+        read -p "Enter new version number: " version
+fi
 
 
 ##############
-## Update install.rdf
+## Update install.rdf and manifest.json
 ##############
 
-perl -pi -e "s/em:version=\"[^\"]*/em:version=\"$version/;" "install.rdf"
-rm "install.rdf.bak"
-git add "install.rdf"
+perl -pi -e "s/em:version=\"[^\"]*/em:version=\"$version/;" src/install.rdf
+perl -pi -e "s/\"version\": \"[^\"]*\"/\"version\": \"$version\"/" src/manifest.json
 
 
 ##############
-## Update update.rdf
+## Create updates.json and update.rdf
 ##############
 
-perl -pi -e "s/<em:version>[^<]*/<em:version>$version/;" \
-          -e "s/<em:updateLink>[^<]*/<em:updateLink>https:\/\/github.com\/UB-Mannheim\/zotero-ocr\/releases\/download\/$version\/zotero-ocr-$version.xpi/;" \
-          -e "s/<em:updateInfoURL>[^<]*/<em:updateInfoURL>https:\/\/github.com\/UB-Mannheim\/zotero-ocr\/releases\/tag\/$version/;" \
-    update.rdf
-git add "update.rdf"
-rm "update.rdf.bak"
+./build.sh "$version"
 
+jq ".addons[\"zotero-ocr@bib.uni-mannheim.de\"].updates[0].version = \"${version}\"" updates.json.tmpl |
```

Comment:
Yes, that line is part of a command chain which writes a new `updates.json`. The input for that comes from the template `updates.json.tmpl`.

Reply:
My code was inspired by https://github.com/zotero/make-it-red/blob/main/make-zips.

Reply:
You can try the release process locally:
- Run `./release.sh 0.7.0`

Then release it on GitHub:
- Run `git push origin 0.7.0`
- Create new release for 0.7.0 on GitHub

Reply:
Okay, I see now that `jq` don't do the changes in-file and therefore a temporary file is needed.

I find these lines a little hard to read and suggest therefore some simplification, like:
```sh
jq '.addons["zotero-ocr@bib.uni-mannheim.de"].updates[0].version = "${version}" | 
    .addons["zotero-ocr@bib.uni-mannheim.de"].updates[0].update_link = "https://github.com/UB-Mannheim/zotero-ocr/releases/tag/${version}/zotero-ocr-${version}.xpi" | 
    .addons["zotero-ocr@bib.uni-mannheim.de"].updates[0].update_hash = "sha256:$(shasum -a 256 build/zotero-ocr-${version}.xpi | cut -d\' \' -f1)"' update-extract.json.tmpl \
   > update.json.tmpl
cp update.json.tmpl update.json
rm update.json.tmpl
```

Moreover, we don't need to commit the temporary file. It should be fine to create this every time and then delete it again.

Reply:
`updates.tmpl` is not a temporary file. It is a template file which is read as input by jq. In your code jq will read from `STDIN` because the first jq command has no file argument.

Reply:
Okay, but we can read this information from the actual `updates,json`. Thus my corrected suggestion would be:
```sh
# update information in update.json using a temporarily file
jq '.addons["zotero-ocr@bib.uni-mannheim.de"].updates[0].version = "${version}" | 
    .addons["zotero-ocr@bib.uni-mannheim.de"].updates[0].update_link = "https://github.com/UB-Mannheim/zotero-ocr/releases/tag/${version}/zotero-ocr-${version}.xpi" | 
    .addons["zotero-ocr@bib.uni-mannheim.de"].updates[0].update_hash = "sha256:$(shasum -a 256 build/zotero-ocr-${version}.xpi | cut -d\' \' -f1)"' update-extract.json.tmpl \
   update.json > update.json.tmpl
cp update.json.tmpl update.json
rm update.json.tmpl
```

Reply:
This modified version might work:
```sh
# update information in update.json using a temporarily file
jq '.addons["zotero-ocr@bib.uni-mannheim.de"].updates[0].version = "${version}" updates.json | 
    .addons["zotero-ocr@bib.uni-mannheim.de"].updates[0].update_link = "https://github.com/UB-Mannheim/zotero-ocr/releases/tag/${version}/zotero-ocr-${version}.xpi" | 
    .addons["zotero-ocr@bib.uni-mannheim.de"].updates[0].update_hash = "sha256:$(shasum -a 256 build/zotero-ocr-${version}.xpi | cut -d\' \' -f1)"' > updates.json.tmp
cp updates.json.tmp updates.json
rm updates.json.tmp
```

It requires five commands instead of three, but does not require an additional template file.

Reply:
jq could also be replaced by perl or sed which allow inline edits, so also can do the job without an additional template or temporary file. But that are cosmetic details without an effect on the resulting plug-in.